src/main/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentService.scala

```diff
@@ -27,6 +28,15 @@ class ConsignmentService(
 
   val maxLimit: Int = config.getInt("pagination.consignmentsMaxLimit")
 
+  def startUpload(startUploadInput: StartUploadInput): Future[Int] = {
+    val now = Timestamp.from(timeSource.now)
+    val consignmentStatusRow = ConsignmentstatusRow(uuidSource.uuid, startUploadInput.consignmentId, "Upload", "InProgress", now)
+    for {
+      _ <- consignmentRepository.addParentFolder(startUploadInput.consignmentId, startUploadInput.parentFolder)
+      res <- consignmentStatusRepository.addConsignmentStatus(consignmentStatusRow)
```

Comment:
How hard would it be to put these into the same transaction? The old status update happened transactionally: https://github.com/nationalarchives/tdr-consignment-api/blob/bc8b36aedb4337d2359964b2edc12fc26307bb15/src/main/scala/uk/gov/nationalarchives/tdr/api/db/repository/FileRepository.scala#L23

Reply:
Yeah makes sense. It's slightly fewer changes because you don't have to pass the `consignmentMetadataRepository` into the `ConsignmentService`. I've added another `addParentFolder` method into `ConsignmentRepository` so it doesn't break the code already calling that. Once we remove the old `addFiles` field, I'll get rid of the old method as well.